### PR TITLE
Adds command to pull whole permission into security context for @Post…

### PR DESCRIPTION
**Pull request by ttomsu**
…Filter optimization. Now a UserPermission with 500 applications will only make 1 network call instead of 500.

Also adds `isAdmin()` endpoint for potential future use.

@duftler @jtk54 @cfieber PTAL.

### CODE REVIEW

File: fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java [hunk 1]
```diff
@@ -83,4 +91,65 @@ private boolean isAuthorized(String username, Resource resource, String resource
...
+      case APPLICATION:
+        return containsAuth.apply(permission.getApplications());
+      default:
+        return false;
+    }
+  }
+
+  @SuppressWarnings("unused")
+  public boolean isAdmin() {
+    return true; // TODO(ttomsu): Chosen by fair dice roll. Guaranteed to be random.
```
Comment (duftler) [inline fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java]: What could go wrong?

**Comment by duftler** [comment]
LGTM.

File: fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java (hunk 1 above)
Comment (jtk54) [inline fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java]: What will actually be used to determine admin privileges (in future)? Certain group membership?

**Comment by jtk54** [comment]
LGTM

File: fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java (hunk 1 above)
Comment (ttomsu) [inline fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java]: I'd imagine so. Everything else is group based, so I'd see no reason why not.

Comment (jtk54) [inline fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatPermissionEvaluator.java]: SGTM

**Pull request merged**